packages/runtime-core/src/helpers/renderSlot.ts

```diff
@@ -38,7 +38,14 @@ export function renderSlot(
       currentRenderingInstance!.parent.isCE)
   ) {
     if (name !== 'default') props.name = name
-    return createVNode('slot', props, fallback && fallback())
+    const vnode = createVNode(
+      'slot',
+      props,
+      fallback && fallback(),
+      fallback ? PatchFlags.BAIL : undefined,
+    )
+    vnode.isSlotFallback = !!fallback
+    return vnode
```

Comment:
I don't understand this part of the change. For custom elements, wouldn't switching between the slot content and the fallback be handled natively by the browser, rather than by the renderer? The presence of a `fallback` doesn't necessarily mean that it is being shown in the component, we render it inside the native `<slot>` either way and let the browser do the rest.

If this is needed then there should probably be a test for it. It seems that the current tests pass even without this change.

Reply:
Updated